Lecture_3/myLib/mylist.c

```diff
@@ -0,0 +1,59 @@
+#include <stdlib.h>
+#include "mylist.h"
+
+TptrList ListAdd(TptrList *link, TptrList elem)
+{
+    if( elem != NULL ){
+	elem->ptrNext=*link;
+	*link=elem;
+    }
+    return elem;
+}
+
+void ListRemove(TptrList *link)
+{
+    TptrList pL=(*link);
+    if( pL != NULL ){
+	*link=pL->ptrNext;
+	free(pL);
+    }
+}
+
+void ListDestroy(TptrList *link)
+{
+    while( (*link) != NULL )
+	ListRemove(link);
+}
+
+TptrList ListCreate(TLISTDATA d)
+{
+    TptrList p;
+    p=malloc(sizeof(struct TList));
+    if( p != NULL ){
+	p->ptrNext=NULL;
+	p->data=d;
+    }
+    return p;
+}
+
+TptrList *ListLast(TptrList *link)
+{
+    if( (*link) != NULL )
+	while(((*link)->ptrNext)!=NULL)
+	    link=&((*link)->ptrNext);
+    return link;
+}
+
+void ListIterate(TptrList head, int (*fdApply)(TLISTDATA *))
+{
+    while((head != NULL) && (fdApply != NULL)){
+	if(fdApply(&(head->data))==0) break;
+	head=head->ptrNext;
+    }
+}
+
+TptrList *ListFind(TptrList *link, int (*criteria)(TLISTDATA,TLISTDATA), TLISTDATA param){
+    while( ((*link) != NULL) && (criteria((*link)->data,param)) )
+	link=&((*link)->ptrNext);
+    return link;
+}
```

Comment:
Try to avoid code(and bugs) duplication.
Use library from previous lecture.

Reply:
I'm thought that different lectures were independent projects. So I added the code (with slight changes) from the previous one **as a library** for the current one.